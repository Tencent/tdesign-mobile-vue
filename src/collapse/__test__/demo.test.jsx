/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import accordionVue from '@/collapse/demos/accordion.vue';
import actionVue from '@/collapse/demos/action.vue';
import baseVue from '@/collapse/demos/base.vue';
import cardVue from '@/collapse/demos/card.vue';
import mobileVue from '@/collapse/demos/mobile.vue';
import placementVue from '@/collapse/demos/placement.vue';

const mapper = {
  accordionVue,
  actionVue,
  baseVue,
  cardVue,
  mobileVue,
  placementVue,
};

describe('Collapse', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Collapse ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
