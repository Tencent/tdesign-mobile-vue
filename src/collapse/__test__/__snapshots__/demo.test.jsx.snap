// Vitest Snapshot v1

exports[`Collapse > Collapse accordionVue demo works fine 1`] = `
<div
  class="accordion-demo"
  data-v-3a4dbde0=""
>
  <div
    class="t-collapse"
    data-v-3a4dbde0=""
    title="手风琴模式"
  >
    
    
    <div
      class="t-collapse-panel t-collapse-panel--active"
      data-v-3a4dbde0=""
      style="height: 0px;"
    >
      <div
        class="t-collapse-panel__header"
      >
        <div
          class="t-collapse-panel__title"
        >
          
          折叠面板标题
          
        </div>
        <div
          class="t-collapse-panel__header-right"
        >
          <!--v-if-->
          <svg
            class="t-icon t-icon-chevron-down t-collapse-panel__header-icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.54 6.46l.92-.92L8 9.08l3.54-3.54.92.92L8 10.92 3.54 6.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-collapse-panel__body"
      >
        
        <div
          class="content"
          data-v-3a4dbde0=""
        >
           此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
        </div>
        
      </div>
    </div>
    <div
      class="t-collapse-panel"
      data-v-3a4dbde0=""
      style="height: 0px;"
    >
      <div
        class="t-collapse-panel__header"
      >
        <div
          class="t-collapse-panel__title"
        >
          
          折叠面板标题
          
        </div>
        <div
          class="t-collapse-panel__header-right"
        >
          <!--v-if-->
          <svg
            class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-collapse-panel__body"
      >
        
        <div
          class="content"
          data-v-3a4dbde0=""
        >
           此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
        </div>
        
      </div>
    </div>
    <div
      class="t-collapse-panel"
      data-v-3a4dbde0=""
      style="height: 0px;"
    >
      <div
        class="t-collapse-panel__header"
      >
        <div
          class="t-collapse-panel__title"
        >
          
          折叠面板标题
          
        </div>
        <div
          class="t-collapse-panel__header-right"
        >
          <!--v-if-->
          <svg
            class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-collapse-panel__body"
      >
        
        <div
          class="content"
          data-v-3a4dbde0=""
        >
           此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
        </div>
        
      </div>
    </div>
    <div
      class="t-collapse-panel t-collapse-panel--disabled"
      data-v-3a4dbde0=""
      style="height: 0px;"
    >
      <div
        class="t-collapse-panel__header"
      >
        <div
          class="t-collapse-panel__title"
        >
          
          折叠面板标题
          
        </div>
        <div
          class="t-collapse-panel__header-right"
        >
          <!--v-if-->
          <svg
            class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-collapse-panel__body"
      >
        
        <div
          class="content"
          data-v-3a4dbde0=""
        >
           此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
        </div>
        
      </div>
    </div>
    
    
  </div>
</div>
`;

exports[`Collapse > Collapse actionVue demo works fine 1`] = `
<div
  class="accordion-demo"
  data-v-5fdf8864=""
>
  <div
    class="t-collapse"
    data-v-5fdf8864=""
  >
    
    <div
      class="t-collapse-panel t-collapse-panel--active"
      data-v-5fdf8864=""
      style="height: 0px;"
    >
      <div
        class="t-collapse-panel__header"
      >
        <div
          class="t-collapse-panel__title"
        >
          
          折叠面板标题
          
        </div>
        <div
          class="t-collapse-panel__header-right"
        >
          <div
            class="t-collapse-panel__header-extra"
          >
            
            收起
            
          </div>
          <svg
            class="t-icon t-icon-chevron-down t-collapse-panel__header-icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.54 6.46l.92-.92L8 9.08l3.54-3.54.92.92L8 10.92 3.54 6.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-collapse-panel__body"
      >
        
        <div
          class="content"
          data-v-5fdf8864=""
        >
           此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
        </div>
        
      </div>
    </div>
    
  </div>
</div>
`;

exports[`Collapse > Collapse baseVue demo works fine 1`] = `
<div
  class="accordion-demo"
  data-v-78207ee9=""
>
  <div
    class="t-collapse"
    data-v-78207ee9=""
  >
    
    <div
      class="t-collapse-panel"
      data-v-78207ee9=""
      style="height: 0px;"
    >
      <div
        class="t-collapse-panel__header"
      >
        <div
          class="t-collapse-panel__title"
        >
          
          折叠面板标题
          
        </div>
        <div
          class="t-collapse-panel__header-right"
        >
          <!--v-if-->
          <svg
            class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-collapse-panel__body"
      >
        
        <div
          class="content"
          data-v-78207ee9=""
        >
           此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
        </div>
        
      </div>
    </div>
    
  </div>
</div>
`;

exports[`Collapse > Collapse mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Collapse 折叠面板
  </h1>
  <p
    class="summary"
  >
    用于对复杂区域进行分组和隐藏 常用于订单信息展示等
  </p>
  <div
    class="tdesign-mobile-demo-block"
    header="01 类型"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      基础折叠面板
    </p>
    
    <div
      class="accordion-demo"
      data-v-78207ee9=""
    >
      <div
        class="t-collapse"
        data-v-78207ee9=""
      >
        
        <div
          class="t-collapse-panel"
          data-v-78207ee9=""
          style="height: 0px;"
        >
          <div
            class="t-collapse-panel__header"
          >
            <div
              class="t-collapse-panel__title"
            >
              
              折叠面板标题
              
            </div>
            <div
              class="t-collapse-panel__header-right"
            >
              <!--v-if-->
              <svg
                class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          <div
            class="t-collapse-panel__body"
          >
            
            <div
              class="content"
              data-v-78207ee9=""
            >
               此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
            </div>
            
          </div>
        </div>
        
      </div>
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      带操作说明
    </p>
    
    <div
      class="accordion-demo"
      data-v-5fdf8864=""
    >
      <div
        class="t-collapse"
        data-v-5fdf8864=""
      >
        
        <div
          class="t-collapse-panel t-collapse-panel--active"
          data-v-5fdf8864=""
          style="height: 0px;"
        >
          <div
            class="t-collapse-panel__header"
          >
            <div
              class="t-collapse-panel__title"
            >
              
              折叠面板标题
              
            </div>
            <div
              class="t-collapse-panel__header-right"
            >
              <div
                class="t-collapse-panel__header-extra"
              >
                
                收起
                
              </div>
              <svg
                class="t-icon t-icon-chevron-down t-collapse-panel__header-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.54 6.46l.92-.92L8 9.08l3.54-3.54.92.92L8 10.92 3.54 6.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          <div
            class="t-collapse-panel__body"
          >
            
            <div
              class="content"
              data-v-5fdf8864=""
            >
               此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
            </div>
            
          </div>
        </div>
        
      </div>
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      手风琴式
    </p>
    
    <div
      class="accordion-demo"
      data-v-3a4dbde0=""
    >
      <div
        class="t-collapse"
        data-v-3a4dbde0=""
        title="手风琴模式"
      >
        
        
        <div
          class="t-collapse-panel t-collapse-panel--active"
          data-v-3a4dbde0=""
          style="height: 0px;"
        >
          <div
            class="t-collapse-panel__header"
          >
            <div
              class="t-collapse-panel__title"
            >
              
              折叠面板标题
              
            </div>
            <div
              class="t-collapse-panel__header-right"
            >
              <!--v-if-->
              <svg
                class="t-icon t-icon-chevron-down t-collapse-panel__header-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.54 6.46l.92-.92L8 9.08l3.54-3.54.92.92L8 10.92 3.54 6.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          <div
            class="t-collapse-panel__body"
          >
            
            <div
              class="content"
              data-v-3a4dbde0=""
            >
               此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
            </div>
            
          </div>
        </div>
        <div
          class="t-collapse-panel"
          data-v-3a4dbde0=""
          style="height: 0px;"
        >
          <div
            class="t-collapse-panel__header"
          >
            <div
              class="t-collapse-panel__title"
            >
              
              折叠面板标题
              
            </div>
            <div
              class="t-collapse-panel__header-right"
            >
              <!--v-if-->
              <svg
                class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          <div
            class="t-collapse-panel__body"
          >
            
            <div
              class="content"
              data-v-3a4dbde0=""
            >
               此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
            </div>
            
          </div>
        </div>
        <div
          class="t-collapse-panel"
          data-v-3a4dbde0=""
          style="height: 0px;"
        >
          <div
            class="t-collapse-panel__header"
          >
            <div
              class="t-collapse-panel__title"
            >
              
              折叠面板标题
              
            </div>
            <div
              class="t-collapse-panel__header-right"
            >
              <!--v-if-->
              <svg
                class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          <div
            class="t-collapse-panel__body"
          >
            
            <div
              class="content"
              data-v-3a4dbde0=""
            >
               此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
            </div>
            
          </div>
        </div>
        <div
          class="t-collapse-panel t-collapse-panel--disabled"
          data-v-3a4dbde0=""
          style="height: 0px;"
        >
          <div
            class="t-collapse-panel__header"
          >
            <div
              class="t-collapse-panel__title"
            >
              
              折叠面板标题
              
            </div>
            <div
              class="t-collapse-panel__header-right"
            >
              <!--v-if-->
              <svg
                class="t-icon t-icon-chevron-up t-collapse-panel__header-icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M12.46 9.54l-.92.92L8 6.92l-3.54 3.54-.92-.92L8 5.08l4.46 4.46z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          <div
            class="t-collapse-panel__body"
          >
            
            <div
              class="content"
              data-v-3a4dbde0=""
            >
               此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容此处可自定义内容可自定义内容 
            </div>
            
          </div>
        </div>
        
        
      </div>
    </div>
    
  </div>
</div>
`;
